### PR TITLE
 Add UI to configure monetization [Publisher] [TYPE/UI]

### PR DESCRIPTION
**Pull request by shilmyhasan**
Add UI to configure monetization in publisher. It provides the ability to enable and disable monetization for an API.

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/BusinessPlans.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                                        id='unchecked.policies.are.not.monetized.click.save.to.monetize'
+                                        defaultMessage='Unchecked polices are not monetized, click save to monetize'
+                                    />
+                                </Typography>
+                            </Grid>
+                        ) : (
+                            <Grid>
+                                <Typography>
+                                    <FormattedMessage
+                                        id='no.commercial.policies.to.monetize'
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/BusinessPlans.jsx]: Hi @shilmyhasan 
Please follow the i18n id naming convention mentioned in [here](https://github.com/wso2/carbon-apimgt/wiki/How-internationalization-(i18n)-works-in-API-Manager-React-Apps#id-naming-convention).

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    }
+
+    /**
+     * Handles the submit action for configuring monetization
+     */
+    handleSubmit() {
+        const { api } = this.props;
+        const properties = this.state.property;
+        const enabled = this.state.monStatus;
+        const body = JSON.stringify({
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: Why do we stringify this JSON object?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+            enabled,
+            properties,
+        });
+        const promisedScopeAdd = api.configureMonetizationToApi(this.props.api.id, body);
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: Shall we give an appropriate name for the variable

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                Alert.info('Something went wrong while configuring monetization');
+                return;
+            }
+            Alert.info('Monetization Configured Successfully');
+        }).catch((error) => {
+            console.error(error);
+            if (error.response) {
+                Alert.error(error.response.body.message);
+            } else {
+                Alert.error('Something went wrong while configuring moneization');
```
Comment (tmkasun) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: Shall we add i18n to all the alert messages?

Comment (tmkasun) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: Here is our developer guide for `i18n`

https://github.com/wso2/carbon-apimgt/wiki/How-internationalization-(i18n)-works-in-API-Manager-React-Apps

Comment (shilmyhasan) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: added in #b29eb6f

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx (hunk 3 above)
Comment (shilmyhasan) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: changed in https://github.com/wso2/carbon-apimgt/pull/6700/commits/3b3e18be4fc8f00a1f3674cd994b28eca44adf3d

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx (hunk 2 above)
Comment (shilmyhasan) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/index.jsx]: changed in b29eb6f

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/BusinessPlans.jsx (hunk 1 above)
Comment (shilmyhasan) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Monetization/BusinessPlans.jsx]: added in b29eb6f

**Comment by tmkasun** [comment]
Hi @shilmyhasan 

Travis build is failing due to an es-lint rule violation, Can you please fix it 

```
ERROR in ./source/src/app/components/Apis/Details/index.jsx
Module build failed (from ./node_modules/eslint-loader/index.js):
Module failed because of a eslint error.

/home/travis/build/wso2/carbon-apimgt/features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/index.jsx
  59:8  error  'Monetization' is defined but never used  no-unused-vars

âœ– 1 problem (1 error, 0 warnings)

 @ ./source/src/app/components/Apis/Apis.jsx 24:0-38 52:33-40 59:33-40
 @ ./source/src/app/ProtectedApp.jsx
 @ ./source/src/Publisher.jsx
 @ ./source/index.jsx
```

~[Full Log](https://api.travis-ci.org/v3/job/572207724/log.txt)

**Pull request merged**